index.d.ts

```diff
@@ -0,0 +1,269 @@
+         * this defaults to `true`, if the data is not a `string`, this will be set to `true`.
+         */
+        binary?: boolean;
+        /**
+         * The last modification date, defaults to the current date.
+         */
+        date?: Date;
+        compression?: string;
+        comment?: string;
+        /** Set to `true` if (and only if) the input is a "binary string" and has already been prepared with a `0xFF` mask. */
+        optimizedBinaryString?: boolean;
+        /** Set to `true` if folders in the file path should be automatically created, otherwise there will only be virtual folders that represent the path to the file. */
+        createFolders?: boolean;
+        /** Set to `true` if this is a directory and content should be ignored. */
+        dir?: boolean;
+
+        /** 6 bits number. The DOS permissions of the file, if any. */
+        dosPermissions?: number | null;
+        /**
+         * 16 bits number. The UNIX permissions of the file, if any.
+         * Also accepts a `string` representing the octal value: `"644"`, `"755"`, etc.
+         */
+        unixPermissions?: number | string | null;
+    }
+
+    interface JSZipObjectOptions {
+        compression: Compression;
+    }
+
+    interface JSZipGeneratorOptions<T extends OutputType = OutputType> {
+        compression?: Compression;
+        compressionOptions?: null | {
+            level: number;
+        };
+        type?: T;
+        comment?: string;
+        /**
+         * mime-type for the generated file.
+         * Useful when you need to generate a file with a different extension, ie: “.ods”.
+         * @default 'application/zip'
+         */
+        mimeType?: string;
+        encodeFileName?(filename: string): string;
+        /** Stream the files and create file descriptors */
+        streamFiles?: boolean;
+        /** DOS (default) or UNIX */
+        platform?: 'DOS' | 'UNIX';
+    }
+
+    interface JSZipLoadOptions {
+        base64?: boolean;
+        checkCRC32?: boolean;
+        optimizedBinaryString?: boolean;
+        createFolders?: boolean;
+    }
+}
+
+interface JSZip {
+    files: {[key: string]: JSZip.JSZipObject};
+
+    /**
+     * Get a file from the archive
+     *
+     * @param Path relative path to file
+     * @return File matching path, null if no file found
+     */
+    file(path: string): JSZip.JSZipObject;
+
+    /**
+     * Get files matching a RegExp from archive
+     *
+     * @param path RegExp to match
+     * @return Return all matching files or an empty array
+     */
+    file(path: RegExp): JSZip.JSZipObject[];
+
+    /**
+     * Add a file to the archive
+     *
+     * @param path Relative path to file
+     * @param data Content of the file
+     * @param options Optional information about the file
+     * @return JSZip object
+     */
+    file<T extends JSZip.InputType>(path: string, data: InputByType[T] | Promise<InputByType[T]>, options?: JSZip.JSZipFileOptions): this;
```

Comment:
On use does one need to write `file<Blob>(...)`, or will it be inferred from the `data` argument?

Reply:
It'll be inferred from the data argument